processor/src/fast/basic_block.rs

```diff
@@ -37,7 +37,7 @@ impl FastProcessor {
         );
 
         // Execute decorators that should be executed before entering the node
-        self.execute_before_enter_decorators(node_id, program, host)?;
+        self.execute_before_enter_decorators(node_id, current_forest, host)?;
```

Comment:
Not for this PR, but we should only execute decorators if we are running in debug mode. Using the something like the API from the pervious comment, this could look like:
```rust
if self.in_debug_mode {
    for decorator in current_forest.get_node_decorators(node_id) {
        self.execute_decorator(decorator, host)?;
    }
}
```